doc/nrf-bm/libraries/bm_storage.rst

```diff
@@ -0,0 +1,105 @@
+.. _lib_storage:
+
+Storage Library
+###############
+
+.. contents::
+   :local:
+   :depth: 2
+
+Overview
+********
+
+The Storage library provides abstractions for reading, writing, and erasing non-volatile memory (NVM).
+It supports multiple storage instances, each bound to a specific memory region, and reports operation results through user-defined event handlers.
+Depending on the backend and runtime state, operations may be synchronous or asynchronous.
+
+Configuration
+*************
+
+The library is enabled and configured entirely using the Kconfig system.
+Set the :kconfig:option:`CONFIG_BM_STORAGE` Kconfig option to enable the library.
+
+Select a storage backend by enabling one of the following Kconfig options:
+
+* :kconfig:option:`CONFIG_BM_STORAGE_BACKEND_RRAM` – RRAM backend. The events reported are synchronous.
+* :kconfig:option:`CONFIG_BM_STORAGE_BACKEND_SD` – SoftDevice backend. The events reported are asynchronous.
+
+SoftDevice backend options:
+
+* :kconfig:option:`CONFIG_BM_STORAGE_BACKEND_SD_QUEUE_SIZE` – Queue size for pending operations.
+* :kconfig:option:`CONFIG_BM_STORAGE_BACKEND_SD_MAX_RETRIES` – Maximum retries if the SoftDevice is busy.
+
+Initialization
+==============
+
+Each storage instance is represented by a :c:struct:`bm_storage` structure.
+
+Initialize an instance with the :c:func:`bm_storage_init` function, providing:
+
+* :c:member:`bm_storage.evt_handler` – Event callback.
+* :c:member:`bm_storage.start_addr` and :c:member:`bm_storage.end_addr` – Accessible address range.
+
+Uninitialize with the :c:func:`bm_storage_uninit` function.
+
+Usage
+*****
+
+Read
+====
+
+Use the :c:func:`bm_storage_read` function to copy data from NVM into RAM.
+The data length must be a multiple of the backend’s program unit and within the configured region.
```

Comment:
"must be a multiple of the backend’s program unit"
This leavs me with the question, what is the backend's program unit

Reply:
In the sample we have the comment 
```
Write buffer size must be a multiple of the program unit. 
To support both RRAM (16 bytes) and SoftDevice (4 bytes) backends, that is 16 bytes.
```
Could probably be added to the documentation.